### PR TITLE
Update randbats/randubs

### PR DESCRIPTION
**Pull request by MacChaeger**
Adds more support for weather/terrain team combos, as well as reevaluating the abilities and items of all previously constructed sets with the updated `teamDetails` whenever a weather/terrain setter is added to the team.

### CODE REVIEW

**Comment by Zarel** [comment]
I don't think this is the right way to re-evaluate teams for weather/terrain.

Maybe we should go through, preliminarily generate a list of Pokémon, and then decide what kind of team it'll be? And then run through the individual `randomSet` functions?

**Comment by MacChaeger** [comment]
That'd work for what I'm doing here since most Pokémon that set weather and terrain with their ability always have that same ability, though Politoed is a but of a problem. I think I can work something out to that effect.

**Comment by Zarel** [comment]
Well, it should still be done at that stage. Decide whether or not Politoed has Drizzle, then decide everything else.

**Comment by MacChaeger** [comment]
Oh fuck I forgot to do Politoed

File: data/random-teams.js [hunk 1]
```diff
@@ -1659,21 +1671,29 @@ class RandomTeams extends Dex.ModdedDex {
...
 			if (set.ability === 'Sand Stream') teamDetails['sand'] = 1;
+			if (set.ability === 'Grassy Surge') teamDetails['grassyTerrain'] = 1;
 			if (set.moves.includes('stealthrock')) teamDetails['stealthRock'] = 1;
 			if (set.moves.includes('toxicspikes')) teamDetails['toxicSpikes'] = 1;
 			if (set.moves.includes('defog') || set.moves.includes('rapidspin')) teamDetails['hazardClear'] = 1;
 		}
 		return pokemon;
 	}
-	randomDoublesSet(template, slot, teamDetails) {
+	randomDoublesSet(template, slot, teamDetails, set) {
```
Comment (Zarel) [inline data/random-teams.js]: Why pass a `set` in?

**Comment by MacChaeger** [comment]
That's just a forgotten leftover from when this was a post-check; I removed it from the singles generator but not doubles, so I'll fix that now.

**Comment by Zarel** [comment]
Looks good to me; @TheImmortal?

File: data/random-teams.js [hunk 2]
```diff
@@ -1210,9 +1210,18 @@ class RandomTeams extends Dex.ModdedDex {
...
 				ability = 'Swift Swim';
 			}
+			if (abilities.includes('Sand Rush') && teamDetails['sand']) {
+				ability = 'Sand Rush';
+			}
+			if (abilities.includes('Slush Rush') && teamDetails['hail']) {
+				ability = 'Slush Rush';
+			}
+			if (abilities.includes('Grassy Pelt') && teamDetails['grassyTerrain']) {
+				ability = 'Grassy Pelt';
```
Comment (Koushien) [inline data/random-teams.js]: Grassy Pelt -> Grass Pelt?

Comment (MacChaeger) [reply data/random-teams.js]: I thought I fixed that. Oops.

**Comment by TheImmortal** [comment]
I get what you're trying to do but I'm not sure it's the best to do it. Let me think about it.

**Comment by MacChaeger** [comment]
Any thoughts so far?

**Comment by TheImmortal** [comment]
I'm not sure forcing the abilities is right. They should naturally get it because weather abilities are decently rated. And if they don't, it means their alternative ability is just as useful to them.

**Comment by MacChaeger** [comment]
Oh, this PR is still open? Guess I actually do have to figure out how to resolve conflicts after all.

**Comment by MacChaeger** [comment]
Oh right, I forgot this doesn't work unless I convince @TheImmortal that Hyper Voice Aurorus is bad, since it's the only Pokemon that doesn't always get its weather setting ability right now, other than the primals, which need to be handled differently so the team generator knows whether they're primal or not before it creates the team's sets.

**Comment by Zarel** [comment]
I still think this is the best approach...

**Comment by TheImmortal** [comment]
It's fine. But, I don't think you should force any abilities.

File: data/random-teams.js [hunk 3]
```diff
@@ -1230,7 +1230,7 @@ class RandomTeams extends Dex.ModdedDex {
 			if (abilities.includes('Prankster') && counter.Status > 1) {
 				ability = 'Prankster';
 			}
-			if (abilities.includes('Swift Swim') && hasMove['raindance']) {
+			if (abilities.includes('Swift Swim') && (hasMove['raindance'] || teamDetails['rain'])) {
```
Comment (TheImmortal) [inline data/random-teams.js]: like this ^

**Comment by TheImmortal** [comment]
Also what is the point of adding Grassy Terrain there?

File: data/random-teams.js (hunk 3 above)
Comment (MacChaeger) [reply data/random-teams.js]: Oh, looks like I missed one.

**Comment by MacChaeger** [comment]
Grass Pelt, but I just noticed that's rated 0.5. With Tapu Bulu existing now, it could probably stand to be rated higher, but considering Sap Sipper is 3.5, Gogoat still wouldn't ever get that ability anyway without it being forced.

**Comment by MacChaeger** [comment]
Also, What did you mean by "It's fine."?

**Comment by Zarel** [comment]
I'm not sure either... TI, can you clarify?

File: data/random-teams.js [hunk 4]
```diff
@@ -1663,6 +1674,13 @@ class RandomTeams extends Dex.ModdedDex {
 			} else if (tier === 'PU') {
 				puCount++;
 			}
+		}
+		while (pokemon.length < 6) {
+			let set = this[this.format.gameType === 'singles' ? 'randomSet' : 'randomDoublesSet'](team.pop(), pokemon.length, teamDetails);
+			pokemon.push(set);
+
+			// Illusion shouldn't be the last Pokemon of the team
+			if (set.ability === 'Illusion' && pokemon.length > 4) pokemon[0] = pokemon.splice(pokemon.length - 1, 1, pokemon[0])[0];
```
Comment (TheImmortal) [inline data/random-teams.js]: This is not a correct thing to do. Pokemon get Focus Sash based on being in slot 0.

**Comment by Zarel** [comment]
This has gone unmaintained for two months.

@TheImmortal what do you want to do with this?

**Comment by TheImmortal** [comment]
I’ve decided that this is unnecessary. There are already complaints about Pokémon getting weather related abilities due to having weather on their team. The general consensus is that it doesn’t work.

**Pull request closed without merging**